### PR TITLE
Add environment specifier support to conda create --file (#15573)

### PR DESCRIPTION
**Pull request by ryanskeith**
## Description

conda create --file now uses the environment specifier plugin system to read environment files (YAML, requirements.txt, explicit, etc.), matching conda env create. Previously it only supported flat package lists (one package per line) like `requirements.txt`.

### Changes
**conda/cli/main_create.py**

- Validate file existence before processing
- Validate that all --file arguments use the same format
- When -n/-p are omitted, read name/prefix from files via `get_name_prefix_from_env_files`
  - Raise a clear error if files do not define name or prefix
- Run clone vs file mutual-exclusivity check before file validation (so --clone + --file yields TooManyArgumentsError instead of EnvironmentFileNotFound)

**conda/models/environment.py**

- Environment.from_cli_with_file_envs: new method that uses get_environment_specifier for each --file instead of the old package-list parsing; returns (Environment, dict[str, Environment]) mapping file path to Environment
- Environment.from_cli: thin wrapper around from_cli_with_file_envs that returns only the merged Environment, keeping previous API promise
- CLI overrides file for name/prefix, also it is necessary to give a name/prefix in the multiple file case
- Environment.merge: name/prefix use last-wins semantics instead of first so it is consistent for all types
  - Except for channels of course which is appending and prioritizes by prepending.
- Environment.merge: external_packages merged by concatenation per key (aligned with other package lists)

**conda/cli/install.py**
- Uses Environment.from_cli_with_file_envs where the file→env map is needed for external package installers

**conda/cli/common.py**

Added:
- get_name_prefix_from_env_files: read name/prefix from env spec files (last file wins) exit early
- validate_environment_files_consistency: ensure all files share the same format
- validate_file_exists: validate file existence before processing

**tests/models/test_environment.py**

- Updated tests as Environment.merge behavior changes
- Updated test as spec_from_url is no longer used

**tests/test_create.py**

- Added simple test to ensure yaml files could be used
- Updated tests against mix packages as the error message was changed
- Added tests for cli behaviors
- Added tests for basic environment spec usage

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda/blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: news/15573-conda-create-env-spec-support [hunk 1]
```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* Add environment specifier plugin support to `conda create --file`. Environment files (YAML, requirements.txt, explicit, etc.) are now parsed via the plugin system, matching `conda env create` behavior. Name and prefix can be read from files when `-n`/`-p` are omitted. (#15573)
```
Comment (travishathaway) [inline news/15573-conda-create-env-spec-support]:
```suggestion
* conda create --file <filename> now supports environment.yaml files as well as requirements.txt and explicit environment exports and also supports automatically detecting name and prefix when --name/--prefix are omitted (#15573)
```

Just trying to put it in simpler language that most users will understand 😄

File: docs/source/user-guide/tasks/creating-projects.rst [hunk 2]
```diff
@@ -63,6 +63,7 @@ Now that we have written a basic ``environment.yml`` file, we can create and act
 from it. To do so, run the following commands::
 
     conda env create --file environment.yml
+    # or: conda create --file environment.yml
```
Comment (travishathaway) [inline docs/source/user-guide/tasks/creating-projects.rst]: I think this should be the preferred method going forward, so it would be nice to flip these around.

File: docs/source/user-guide/tasks/manage-environments.rst [hunk 3]
```diff
@@ -108,6 +108,10 @@ Use the terminal for the following steps:
 
       conda env create -f environment.yml
 
+   Alternatively, ``conda create --file environment.yml`` accepts the same
```
Comment (travishathaway) [inline docs/source/user-guide/tasks/manage-environments.rst]: Same thing here. I think we should promote `conda create --file <file>` as the recommended way to do this going forward with a note that `conda env create --file <file>` is still possible to support backwards compatibility.

File: conda/cli/install.py [hunk 4]
```diff
@@ -443,6 +444,35 @@ def install(args, parser, command="install"):
 
     handle_txn(unlink_link_transaction, prefix, args, newenv)
 
+    if (
+        newenv
```
Comment (soapy1) [inline conda/cli/install.py]: What is the motivation for including this `newenv` check here? I would expect `conda update` or `conda install` to deal with updates to the `pip` section of an environment file. For example, I would expect the following workflow to work:

1. create a new environment.yaml
```
# environment.yaml
name: pip
dependencies:
  - pip
  - numpy
  - pip:
    - scipy
```

2. create the environment with conda
```
$ conda create -p /tmp/test --file environment.yaml
...
# environment created + pip deps installed
```

3. update the environment.yaml
```
# environment.yaml
name: pip
dependencies:
  - pip
  - numpy
  - pip:
    - scipy
    -  pandas   # new dep
```

4. update the environment with conda
```
$ conda install -p /tmp/test --file environment.yaml
. . .
```
**I would expect** the outcome of step 4 to be  that the new pip packages (pandas) are installed. However the **actual behaviour** is that these changes are ignored.

File: conda/cli/install.py [hunk 5]
```diff
@@ -443,6 +444,35 @@ def install(args, parser, command="install"):
...
+        and env.external_packages
+        and not context.dry_run
+        and not context.download_only
+    ):
+        from .. import CondaError
+        from ..env.installers.base import get_installer
+        from ..env.pip_util import get_pip_workdir
```
Comment (soapy1) [inline conda/cli/install.py:450]: This is a neat way to isolate the pip installer stuff! Hopefully, we'll get to replace this soon 🌈

File: conda/cli/install.py [hunk 6]
```diff
@@ -443,6 +444,35 @@ def install(args, parser, command="install"):
...
+
+        for installer_type, pkg_specs in env.external_packages.items():
+            try:
+                installer = get_installer(installer_type)
+                if installer_type == "pip":
```
Comment (soapy1) [inline conda/cli/install.py]: There is certainly room for improvement in the environment model's `external_packages` bit. It can maybe be removed in the future as conda starts to adopt conda-pypi's native support for pypi packages.

In the mean time, there is this [`conda.models.environment.EXTERNAL_PACKAGES_PYPI_KEY`](https://github.com/conda/conda/blob/main/conda/models/environment.py#L45) if you choose to want to use it.

```suggestion
                if installer_type == EXTERNAL_PACKAGES_PYPI_KEY:
```

File: docs/source/user-guide/tasks/manage-environments.rst (hunk 3 above)
Comment (ryanskeith) [reply docs/source/user-guide/tasks/manage-environments.rst]: My brain prefers `conda env` but I will change these around.

File: news/15573-conda-create-env-spec-support (hunk 1 above)
Comment (ryanskeith) [reply news/15573-conda-create-env-spec-support]: I tend to be very technical.  Thanks for edit.

File: conda/cli/main_create.py [hunk 7]
```diff
@@ -96,41 +96,56 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
...
+    validate_environment_files_consistency(args.file)
+
+    if not args.name and not args.prefix and args.file:
+        name, prefix = get_name_prefix_from_env_files(args.file)
+        if name is not None:
+            args.name = name
+        if prefix is not None and args.name is None:
+            args.prefix = prefix
+        if args.name is not None or args.prefix is not None:
+            context.__init__(argparse_args=args)
```
Comment (soapy1) [inline conda/cli/main_create.py:144]: Whats the motivation for re-intializing the context here?

File: conda/cli/main_create.py [hunk 8]
```diff
@@ -96,41 +96,56 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
...
+            list(args.file),
+            "`--file` and `--clone` arguments are mutually exclusive.",
+        )
+
+    for fpath in args.file:
+        validate_file_exists(fpath)
+    validate_environment_files_consistency(args.file)
+
+    if not args.name and not args.prefix and args.file:
+        name, prefix = get_name_prefix_from_env_files(args.file)
```
Comment (soapy1) [inline conda/cli/main_create.py]: What is the intended behaviour for getting the right name/prefix when multiple environment files with a name or prefix is set?

Currently, if I have 2 environment files conda will choose the name/prefix from the last file. So, for the 2 environments files
```
prefix: /tmp/sdg
dependencies:
  - pip
  - numpy
  - pip:
    - scipy
```

```
name: nlp
dependencies:
  - nltk
```

If I run `conda create  --file tests/env/support/with-pip.yml --file tests/env/support/simple.yml` or `conda create --file tests/env/support/simple.yml  --file tests/env/support/with-pip.yml`, I'll get a different install location. To me, that is kind of confusing. It might be less confusing if conda produces and error requesting that the user provides a `--name`/`--prefix` arg if more than one `--file` is provided?

Some further QA here:
1. Also to confirm that if I set `--name` or `--prefix` these arguments are respected and take precedence over the name/prefix in the environment files.
2. And if I use files that do not provide a name/prefix I get an error asking to specify one. For example
```
$ conda create  --file tests/requirements-benchmarks.txt  --file tests/requirements-ci.txt  

ArgumentError: The environment file(s) do not specify a name or prefix. Please provide one via -n/--name or -p/--prefix.
```

File: conda/env/installers/pip.py [hunk 9]
```diff
@@ -27,33 +28,28 @@ def _pip_install_via_requirements(prefix, specs, args, *_, **kwargs):
...
-    url_scheme = args.file.split("://", 1)[0]
-    if url_scheme in CONDA_SESSION_SCHEMES:
-        pip_workdir = None
-    else:
-        try:
-            pip_workdir = op.dirname(op.abspath(args.file))
-            if not os.access(pip_workdir, os.W_OK):
-                pip_workdir = None
-        except AttributeError:
-            pip_workdir = None
```
Comment (soapy1) [inline conda/env/installers/pip.py:40]: hmmm, I think this is a pretty significant change in expectation for this function. Previously, it would work out what the `pip_workdir` is. But this change puts that responsibility on the caller. This might cause unintended consequences to users of the `conda.env.installer.pip.install` function who have not been notified of this change.

Would you be open to including some default value here. Like:
```
if workdir is None:
  workdir = get_pip_workdir(args.file)
```

File: conda/env/pip_util.py [hunk 10]
```diff
@@ -14,11 +14,35 @@
...
+    """
+    Derive the working directory for pip install when resolving relative paths
+    in requirements (e.g. -e ./local_pkg).
+
+    Returns None for URLs or when no usable path is provided.
+    """
+    if file_path is None:
+        return None
+    if isinstance(file_path, list):
+        file_path = file_path[0] if file_path else None
```
Comment (soapy1) [inline conda/env/pip_util.py]: Hmmm, I don't think it's sufficient to pull the first file path and make that the pip workdir. The pip section of the environment.yaml is meant to support relative paths. For example, https://github.com/conda/conda/blob/main/tests/env/support/advanced-pip/env_template.yml. 

So, if I run the command:
```
$ conda create -p /tmp/six --file tests/env/support/with-pip.yml  --file  tests/env/support/advanced-pip/env_template.yml
```

conda will run the pip install with the workdir for the first file `tests/env/support/with-pip.yml` and will end up with the error:
```
Pip subprocess error:
ERROR: ./module_to_install_in_editable_mode is not a valid editable requirement. It should either be a path to a local project or a VCS URL (beginning with bzr+http, bzr+https, bzr+ssh, bzr+sftp, bzr+ftp, bzr+lp, bzr+file, git+http, git+https, git+ssh, git+git, git+file, hg+file, hg+http, hg+https, hg+ssh, hg+static-http, svn+ssh, svn+http, svn+https, svn+svn, svn+file).

failed
```

This is also confusing because it takes the first file path. But in `get_name_prefix_from_env_files` we take the data from the last file path.

File: conda/models/environment.py [hunk 11]
```diff
@@ -296,26 +295,24 @@ def merge(cls, *environments):
         prefixes = [env.prefix for env in environments if env.prefix]
 
         if names:
-            name = names[0]
+            name = names[-1]
```
Comment (soapy1) [inline conda/models/environment.py:320]: What's the motivation for chaning this to last taking precedence over first?

File: conda/models/environment.py [hunk 12]
```diff
@@ -554,6 +560,11 @@ def from_cli(
             config=EnvironmentConfig.from_context(),
         )
 
+        if envs_from_file:
+            file_env = cls.merge(*envs_from_file)
+            return cls.merge(file_env, cli_env)
```
Comment (soapy1) [inline conda/models/environment.py]: hmm, I'm not quite sure about the merging of config that happens here. 
This currently does not respect the precedence outlined in https://docs.conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html, instead it puts the environment file as the highest precedence source of configuration. But, it should probably be at least below command line parameters (in terms of precedence). Currently this is only effecting the `channels` config, since this is the only piece of config exposed in an environment file format.

Also note, we have a similar situation with the `conda env create` command. But in that case, users can't provide a `--channels` cli argument and channels are all appended to eachother so this doesn't become such an issue.

Consider the example setup:

**condarc**
```
channels:
  - defaults
```

**simple.yml**
```
name: ex
channels:
  - conda-forge
dependencies:
  - python
  - r
```

## case 1: run a normal conda install
**expect**: 
  1. to get python installed from the conda-forge channel
  2. to get r installed from the conda-forge channel

**actual**: 
  1. ✔️ I get python installed from conda-forge
  2.  ✔️ I get r installed from the conda-forge channel
```
$ conda create -p /tmp/eight --file simple.yml  --dry-run  
Channels:
 - conda-forge
 - defaults
 . . .
  python             conda-forge/linux-64::python-3.14.3-h32b2ec7_101_cp314 
  r                  conda-forge/noarch::r-4.5-r45hd8ed1ab_1009 
. . .
```

## case 2: run a conda install with a `--channel r`
**expect**: 
1. to get python installed from the conda-forge channel
2. to get r installed from the r channel
3.  have the r channel be the highest priority

**actual**: 
1. ✔️ I get python installed from conda-forge,
2. ❌ I get r installed from the conda-forge channel
3. ❌ I get conda-forge as the highest priority channel
```
$ conda create -p /tmp/eight --file simple.yml --channel r  --dry-run  
Channels:
 - conda-forge
 - r
 - defaults
 . . .
python             conda-forge/linux-64::python-3.14.3-h32b2ec7_101_cp314 
r                  conda-forge/noarch::r-4.5-r45hd8ed1ab_1009 ```
. . .
```

ref: https://github.com/conda/conda/pull/15199

File: conda/models/environment.py [hunk 13]
```diff
@@ -577,7 +588,7 @@ def extrapolate(self, platform: str) -> Environment:
             with repodata_manager as repodata_fn:
                 solver = solver_backend(
                     prefix="/env/does/not/exist",
-                    channels=context.channels,
+                    channels=self.config.channels,
```
Comment (soapy1) [inline conda/models/environment.py:646]: I'm not too familar with this `extrapolate` function. What's the motivation for changing this? If this is a required change for some other bug can we do it in a seperate PR to keep this one focused on just adding env spec support to `conda create`.

File: conda/cli/main_create.py [hunk 14]
```diff
@@ -96,41 +96,56 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
...
+            len(args.file),
+            list(args.file),
+            "`--file` and `--clone` arguments are mutually exclusive.",
+        )
+
+    for fpath in args.file:
+        validate_file_exists(fpath)
+    validate_environment_files_consistency(args.file)
+
+    if not args.name and not args.prefix and args.file:
```
Comment (soapy1) [inline conda/cli/main_create.py]: Is this logic also now applicable to `conda install`?

For example, if I have an environment.yaml
```
prefix: /tmp/myenv
dependencies:
  - numpy
``` 

And I run a `conda install --file environment.yaml` should I expect it to create an environment `/tmp/myenv`? Or should it try to install the specified dependencies into the current environment?

**Review by soapy1 — DISMISSED** [review]
Looking great so far! I have a few questions around what the intended behaviour of some edge cases should be.

It would also be nice to add a few more tests. It might be neat to parameterize the integration tests in https://github.com/conda/conda/blob/main/tests/env/test_create.py to run against `conda env create` and `conda create` (I'm just spitballing there).

File: conda/cli/install.py (hunk 4 above)
Comment (ryanskeith) [reply conda/cli/install.py]: Ah, hyper focused on the `create` part of the ticket.  Totally correct on pointing it out for the other functionality.

File: conda/cli/install.py (hunk 5 above)
Comment (ryanskeith) [reply conda/cli/install.py:450]: I didn't like how I needed to do this.  I felt that the better alternative would be a refactor of the env.installers workflow.  It would be interesting to see if we allowed the installer to know what to do.  It seemed like too much change to have it scale better.  Might be worth an exploratory demo.

File: conda/cli/main_create.py (hunk 7 above)
Comment (ryanskeith) [reply conda/cli/main_create.py:144]: The motivation is the same as the --dry-run.  We need to reinit the context if the code changes -name or -prefix. In the dry-run case, a temp path is supplied. 

At this point the determine_target_prefix has been run in the main.py code.  If name or prefix is not set, the target prefix is the base environment.  If one of them is set it is the normal behavior at the command line the target_prefix has already been set.  This case captures the case where an environment file has that information and the users did not supply on the cli.

File: conda/env/installers/pip.py (hunk 9 above)
Comment (ryanskeith) [reply conda/env/installers/pip.py:40]: That's a great idea.  One of my worries was if a deprecation notice was needed.  I like this idea.  I will include it.

File: conda/models/environment.py (hunk 11 above)
Comment (ryanskeith) [reply conda/models/environment.py:320]: Other parts of the merge code overwrote (configs and variables were last key wins).  I thought name and prefix should be the same as well for consistency.

File: conda/models/environment.py (hunk 13 above)
Comment (ryanskeith) [reply conda/models/environment.py:646]: Ah, I was running some code locally and saw that that my local .condarc was affecting the tests.  After consulting another conda team member and determined it was a bug in the code I made this small change.  I do understand that a separate PR is another option BUT this was also after recently reading the HOW WE USE GITHUB guide.

https://github.com/conda/conda/blob/main/HOW_WE_USE_GITHUB.md#should-we-make-unrelated-changes-at-the-same-time

This states it is ok to make small changes to unrelated work in PR's.  So I went ahead and included it and moved on.

File: conda/models/environment.py (hunk 12 above)
Comment (ryanskeith) [reply conda/models/environment.py]: You are totally right that the CLI should take precedence and that is the intention of the code.  I changed the code so that the last should take precedence.  It seems the intention of the EnvironmengtConfig class is the same. 

https://github.com/conda/conda/blob/3c53f03c80c3645f22e6ec69789a3902308b7396/conda/models/environment.py#L105-L115

I will look more into this to understand why it is not doing this.

File: conda/env/pip_util.py (hunk 10 above)
Comment (ryanskeith) [reply conda/env/pip_util.py]: I was thinking more of the case where we have all the requirements.txt in the same directory (like conda).  

You are bringing up a case where we need to do multiple pip installs.  Let's say that I have multiple files in different locations and they all have relatives paths taking one of these files will not lead to a good install.  It seems the code needs to be smarter if we allow for multiple files.

File: conda/cli/install.py (hunk 6 above)
Comment (ryanskeith) [reply conda/cli/install.py]: For now, I feel we can hold off on this change as we are thinking about how we want external packages to be handled.

File: conda/cli/main_create.py (hunk 8 above)
Comment (ryanskeith) [reply conda/cli/main_create.py]: That is correct.  So you are saying that if one is supplied we are ok and just use it but if both are then ambiguity exists and we should ask the user for help.  As a power user, I do like the idea that I can make use priority to my advantage and save typing.  I do see your point that this is a new case and the user should be explicit here.

I am honestly torn on this issue.

File: conda/env/pip_util.py (hunk 10 above)
Comment (ryanskeith) [reply conda/env/pip_util.py]: So after reading the CEP on requirements.txt, conda expects requirements.txt to have their paths based on the cwd.

Moving forward with CEP 24 and handling multiple pip sections, the requirement is different.  We will need to loop through these for now as the expected behavior is for pip to run based off the environment file.

File: conda/cli/main_create.py (hunk 14 above)
Comment (ryanskeith) [reply conda/cli/main_create.py]: That's a great question. 

As a user, I think it would make the most sense to me that it would install in the current environment.  conda install is first and foremost a package command (not environment).  At least in my mind.

File: conda/cli/main_create.py [hunk 15]
```diff
@@ -96,41 +96,61 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
...
+            "`--file` and `--clone` arguments are mutually exclusive.",
+        )
+
+    for fpath in args.file:
+        validate_file_exists(fpath)
+    validate_environment_files_consistency(args.file)
+
     if not args.name and not args.prefix:
-        if context.dry_run or context.download_only:
+        if args.file and len(args.file) > 1:
```
Comment (ryanskeith) [inline conda/cli/main_create.py:129]: I got some feedback that for multiple files that we should always ask the user for a name or prefix to keep things simple.  For the single case, we will look for them.

File: conda/env/pip_util.py (hunk 10 above)
Comment (ryanskeith) [reply conda/env/pip_util.py]: Resolved by removing list behavior altogether in this function.  Since we are looping by file now in `conda create` we will call this over and over again if need be.

File: conda/cli/main_create.py (hunk 8 above)
Comment (soapy1) [reply conda/cli/main_create.py]: Looks like you went with asking the user to provide a -n/-p if there are multiple files that define an install location. That's great!

I can also confirm that this doesn't change the behviour for users running commands like
```
$ conda create -p /something --file requirements.txt --file requirements2.txt

$ conda install -p /something --file requirements.txt --file requirements2.txt
``` 

@ryanskeith could you also add a few tests for this new functionality :pray:

File: conda/env/pip_util.py (hunk 10 above)
Comment (soapy1) [reply conda/env/pip_util.py]: Nice solution! I think not ideal to have to call pip multiple times. But when I was testing it out it didn't feel out of place or anything. So I think this works well.

File: conda/models/environment.py (hunk 12 above)
Comment (soapy1) [reply conda/models/environment.py]: hmmm, I think this still isn't quite right now I get the follwoing behaviour:

## case 1: run a normal conda install
**expect**: 
  1. to get python installed from the conda-forge channel
  2. to get r installed from the conda-forge channel
  3. conda-forge is the highest priority channel

**actual**: 
  1.  ❌ I get python installed from defaults
  2.  ❌ I get r installed from the defaults channel
  3.  ❌ defaults is the highest priority channel

```
$ conda create -p /tmp/eight --file simple.yml  --dry-run  
Channels:
 - defaults
 - conda-forge
 . . .
  python             pkgs/main/linux-64::python-3.14.3-h490e9c7_101_cp314  
 r                  pkgs/r/linux-64::r-4.3.1-r43_0 . . .
```

## case 2: run a conda install with a `--channel r`
**expect**: 
1. to get python installed from the conda-forge channel
2. to get r installed from the r channel
3.  have the r channel be the highest priority

**actual**: 
1. ❌ I get python installed from defaults,
2. ✔️ I get r installed from the r channel
3. ✔️ I get r as the highest priority channel
```
$ conda create -p /tmp/eight --file simple.yml --channel r  --dry-run  
Channels:
 - r
 - defaults
 - conda-forge
 . . .
  python             pkgs/main/linux-64::python-3.14.3-h490e9c7_101_cp314 
  r                  r/linux-64::r-4.3.1-r43_0 
. . .
```

File: conda/cli/main_create.py (hunk 14 above)
Comment (soapy1) [reply conda/cli/main_create.py]: That sounds sensible to me :+1:

File: conda/cli/install.py [hunk 16]
```diff
@@ -443,6 +444,36 @@ def install(args, parser, command="install"):
...
+                    else:
+                        installer.install(prefix, list(pkg_specs), args, file_env)
+                except InvalidInstaller:
+                    raise CondaError(
+                        f"Unable to install package for {installer_type} from environment file {fpath}. "
+                        "Please ensure your dependencies file has the correct spelling."
+                    )
+
+        if env.variables:
+            PrefixData(prefix).set_environment_env_vars(env.variables)
```
Comment (soapy1) [inline conda/cli/install.py]:
```suggestion
    if env.variables :
        PrefixData(prefix).set_environment_env_vars(env.variables)
```

Looks like this is indented a tiny bit too much.  Currently env vars don't get exported. For example a env.yaml that looks like
```
variables:
  VARIABLE: value
  VARIABLE2: value2
dependencies: []
```

File: conda/models/environment.py (hunk 12 above)
Comment (soapy1) [reply conda/models/environment.py]: Cool, this seems to work now! I also did some further testing with overriding channels.

Consider the environment file
```
# simple.yml
name: ex
dependencies:
  - python
  - r
```

## case 1 run a create

**expect**
* packages to come from the default channel
* only the default channel is used

**actual**
* :heavy_check_mark: packages from the default channel
* :heavy_check_mark:  only the default channel is used

```
 $ conda create -p /tmp/eight --file simple.yml  --dry-run

Channels:
 - defaults
Platform: linux-64
. . .
```

## case 2 use conda-forge channel
**expect**
* packages to come from the conda-forge channel primarily
* the conda-forge and default channels are used

**actual**
* :heavy_check_mark: packages from the conda-forge channel
* :heavy_check_mark:   the conda-forge and default channels are used

```
$ conda create -p /tmp/eight --file simple.yml  --dry-run   --channel conda-forge                    
Channels:
 - conda-forge
 - defaults
Platform: linux-64
 . . .
```


## case 3 override channels

**expect**
* packages to come from the conda-forge channel
* only the conda-forge channel is used

**actual**
* :heavy_check_mark: packages from the conda-forge channel
* :heavy_check_mark:  only the conda-forge channel is used

```
 $ conda create -p /tmp/eight --file simple.yml  --dry-run   --channel conda-forge --override-channels

Channels:
 - conda-forge
 Platform: linux-64
. . .
```

File: conda/models/environment.py [hunk 17]
```diff
@@ -487,7 +507,7 @@ def from_cli(
         cls,
         args: Namespace,
         add_default_packages: bool = False,
-    ) -> Environment:
+    ) -> tuple[Environment, dict[str, Environment]]:
```
Comment (soapy1) [inline conda/models/environment.py]: I think this api change is pretty significant. How would you feel about keeping `from_cli` the same as it was and intorduce a new function that returns this tuple of `[Environment, dict[str, Environment]]`?

File: conda/models/environment.py [hunk 18]
```diff
@@ -533,27 +553,55 @@ def from_cli(
...
 
+        if envs_from_file:
+            file_env = cls.merge(*envs_from_file)
+            merged = cls.merge(base_env, file_env, cli_env)
+
+        else:
+            merged = cls.merge(base_env, cli_env)
+
+        # Respect override_channels flag and only use channels from the CLI
+        if getattr(args, "override_channels", False):
```
Comment (soapy1) [inline conda/models/environment.py:617]: I'm not a super big fan of this special management of configured channels (or the environemnt config merging) situation in general. It would be better if all the logic for merging configurations from all sources happened in the context object as suggested in https://github.com/conda/conda-planning/issues/88. I think that's a pretty big piece of work and I wouldn't want to explode the scope of this PR. 
Would a reasonable compromise be to create a to do item/issue for addressing this?

File: conda/models/environment.py (hunk 17 above)
Comment (ryanskeith) [reply conda/models/environment.py]: I felt that because the internal code only calls it once, it was safe to make this change. I will look at what it would look like to make a new function.

File: conda/models/environment.py (hunk 18 above)
Comment (ryanskeith) [reply conda/models/environment.py:617]: I agree it would be very reasonable.  I will make todo note about channel merging happening in a central place.  Perhaps we could also find some time before Q3 planning to chart out what that would look like for next steps.

File: conda/models/environment.py (hunk 17 above)
Comment (ryanskeith) [reply conda/models/environment.py]: After looking at it I realize there is a simple solution.  I will keep from_cli the same and it is a thin wrapper around a new function called from_cli_with_file_envs.

File: docs/source/dev-guide/plugins/environment_specifiers.rst [hunk 19]
```diff
@@ -46,7 +46,9 @@ Be sure to be very specific when implementing the ``can_handle`` method. It shou
 return a ``True`` if the file can be parsed by the plugin. Making the ``can_handle``
 method too permissive in the types of files it handles may lead to conflicts with other
 plugins. If multiple installed plugins are able to ``can_handle`` the same file type,
-conda will return an error to the user.
+conda will return an error to the user.  Also, assure that the ``can_handle`` method
```
Comment (travishathaway) [inline docs/source/dev-guide/plugins/environment_specifiers.rst]:
```suggestion
conda will return an error to the user.  Also, ensure that the ``can_handle`` method
```

**Comment by travishathaway** [comment]
@soapy1,

I think it would make sense for you to take another quick look at the pull request. You can specifically focus on the changes I introduced in my last commit. Below is an explanation of what/why I changed things.

### Fixing `prefix` defaults

While I was going through and QA testing this myself, I discovered that this did not produce the expected error:

> When -n/-p are omitted, read name/prefix from files via get_name_prefix_from_env_files 

I was testing it with the command:

```
conda create --file env.yaml
```

Instead, it was giving the following error:

```
CondaValueError: 'base' is a reserved environment name
```

This was happening because of this line:

https://github.com/conda/conda/blob/aa6e9eca06d392ef8e6408369aa7400d5b9a031e/conda/env/env.py#L442

So, to fix this, I have simply removed falling back to `context.target_prefix`. I think this is the right behavior because supplying that as a fall back should be done outside of the object's instantiation (e.g. be provided directly as the `prefix` argument to the `EnvironmentYaml` object itself).

### Command line parsing

The other changes I made were related to command line parsing. I also decided to change the `get_name_prefix_from_env_files` function to just work on single env files because in the `main_create.py` file in the `execute` function, we're not actually passing in multiple env files to determine name/prefix (we raise an error before getting to that point).
